change/@ni-nimble-blazor-1cbfc49b-865a-4368-ab38-15ee937d0420.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Added an additional canvas in wafer map template caused the blazor tests to fail due to missing unique identifiers for canvases, adding different ids for the canvases and explicitly telling the tests to look for one canvas id solved the issue",
```

Comment:
@Razvan1928 I did not get a chance to finish reviewing this today but expect to look closer tomorrow. Some of the async setup patterns I think need to be looked at a bit more closely. In particular passing a Promise to Dom.queueUpdate seems strange as [queueUpdate is not async aware](https://github.com/microsoft/fast/blob/archives/fast-element-1/packages/web-components/fast-element/src/dom.ts#L55). I don't have any concrete suggestions yet but that is the area I'm concerned about.

Reply:
Solved in this commit https://github.com/ni/nimble/pull/1929/commits/d1dd606b21eb59b19f9a30d6eb5771ed64effd25